src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsExtensionBuilderExtensions.cs

```diff
@@ -38,7 +42,7 @@ public static class WebJobsExtensionBuilderExtensions
         /// <param name="builder">The <see cref="IWebJobsExtensionBuilder"/> to configure.</param>
         /// <param name="configure">The <see cref="Action{IConfigurationSection, TOptions}"/> that will be invoked when configuring the options.</param>
         /// <returns>The configured <see cref="IWebJobsExtensionBuilder"</returns>
-        public static IWebJobsExtensionBuilder ConfigureOptions<TOptions>(this IWebJobsExtensionBuilder builder, Action<IConfigurationSection, TOptions> configure) where TOptions : class
+        public static IWebJobsExtensionBuilder ConfigureOptions<TOptions>(this IWebJobsExtensionBuilder builder, Action<IConfigurationSection, TOptions> configure) where TOptions : class, new()
```

Comment:
I think the addition of new() might be potentially breaking here? I think all options classes already have to have default constructors (i.e. if they didn't they already wouldn't work) but technically this would be a breaking change I think to the methods, so if not necessary I'd say we omit this.

Reply:
Yes. This would need to be evaluated against the consumers as it is a constraint addition and would be breaking if violated.

Reply:
@TsuyoshiUshio can you try updating this to use `IOptionsMonitor` as opposed to `IOptions` here?

Reply:
Done!